src/passes/pass.cpp

```diff
@@ -138,19 +138,23 @@ void PassRunner::addDefaultGlobalOptimizationPasses() {
 }
 
 void PassRunner::run() {
-  if (options.debug) {
+  // BINARYEN_PASS_DEBUG is a convenient commandline way to log out the toplevel passes, their times,
+  //                     and validate between each pass.
+  //                     (we don't recurse pass debug into sub-passes, as it doesn't help anyhow and
+  //                     also is bad for e.g. printing which is a pass)
+  static const int passDebug = getenv("BINARYEN_PASS_DEBUG") ? atoi(getenv("BINARYEN_PASS_DEBUG")) : 0;
+  if (!isNested() && (options.debug || passDebug)) {
```

Comment:
Basically, it does seem to make sense to have the information of "run a set of passes" vs "I am a pass, and I want to run some internal passes". And a nice API for that is "create a runner for nested passes".

I agree if we add another type of pass runner then there could be a problem. I don't see a reason for that, but who knows, yeah.

Instead, are you suggesting that whether we are nested or not would be a property, and specified in the constructor or such?